### PR TITLE
[DO NOT MERGE] Experimental codes to split fwd conv2d into smaller ops.

### PR DESCRIPTION
**Pull request by whchung**
Experimental codes. DO NOT MERGE.

Show case how we could split `miopen.conv2d` into several kernel invocations including `miopen.conv2d` + `miopen.conv2d_dummy`.

The infrastructure in this PR would eventually be evolved to split `miopen.conv2d_bwd_data` into multiple pieces.

This PR depends on #137 .

One pass, `-miopen-split-conv2d` was added and introduced to the beginning of the pipeline.

Compilation part is finished. Changes needed in `mlir-rocm-runner` is undergoing. Unit tests ARE EXPECTED TO FAIL FOR NOW.

### CODE REVIEW

**Comment by sjw36** [comment]
Just curious, why have a different module for each gpu kernel?

File: mlir/lib/Conversion/MIOpenToGPU/MIOpenToGPU.cpp
Comment (sjw36) [inline mlir/lib/Conversion/MIOpenToGPU/MIOpenToGPU.cpp:163]: I know this is a small list but using a map would make it look cleaner.

**Comment by whchung** [comment]
> Just curious, why have a different module for each gpu kernel?

Good observation. This is to tame lower-level passes in mlir-rocm-runner. I haven’t dived into details yet but I guess it’s the pass which gets the compiled ISA binary as string attribute which needs this

**Review by sjw36 — APPROVED** [review]

File: mlir/lib/Conversion/MIOpenToGPU/MIOpenToGPU.cpp
Comment (whchung) [reply mlir/lib/Conversion/MIOpenToGPU/MIOpenToGPU.cpp:163]: this part of the logic was written very hastily (in < 15 minutes?). indeed a map is more desirable.

**Comment by whchung** [comment]
I will hold this PR for now and:
- [ ] populate dummy ops per `stride/dilation` attribute, instead of hard-coded 3 right now.
- [ ] change `convert-mopen-to-gpu` interface so it's possible just to lower one kernel at a time to fulfill what's needed from MIOpen.
- [ ] wait for @asleepzzz and @ltqin come up with the reduced version of v4r1 `mopen.conv2d_bwd_data`, where cases with `stride=dilation=1` can pass, then revise this PR to accommodate necessary changes needed for v4r1.

**Comment by whchung** [comment]
Revisit this after #174 gets merged.

**Pull request closed without merging**
